### PR TITLE
Handle new capabilities gracefully

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/basic/cap-list.c [hunk 1]
```diff
@@ -36,7 +36,7 @@ int capability_from_name(const char *name) {
         /* Try to parse numeric capability */
         r = safe_atoi(name, &i);
         if (r >= 0) {
-                if (i >= 0 && (size_t) i < ELEMENTSOF(capability_names))
+                if (i >= 0 &&)
```
Comment (panaman67) [inline src/basic/cap-list.c:39]:
```suggestion
                if (i >= 0)
```

File: src/basic/cap-list.c [hunk 2]
```diff
@@ -64,11 +65,14 @@ int capability_set_to_string_alloc(uint64_t set, char **s) {
...
                 if (set & (UINT64_C(1) << i)) {
                         const char *p;
+                        char buf[2 + 16 + 1];
                         size_t add;
 
                         p = capability_to_name(i);
-                        if (!p)
-                                return -EINVAL;
+                        if (!p) {
+                                xsprintf(buf, "0x%lx", i);
```
Comment (poettering) [inline src/basic/cap-list.c:71]: `i` is an `int`, hence just use `%x` rather than `%lx`here?

File: src/basic/cap-list.c [hunk 3]
```diff
@@ -36,7 +37,7 @@ int capability_from_name(const char *name) {
         /* Try to parse numeric capability */
         r = safe_atoi(name, &i);
         if (r >= 0) {
-                if (i >= 0 && (size_t) i < ELEMENTSOF(capability_names))
+                if (i >= 0 &&)
                         return i;
```
Comment (poettering) [inline src/basic/cap-list.c:40]: hmm, given that we store caps masks in a 64 bit unsigned integer we should refuse caps > 63 here

**Comment by keszybz** [comment]
Oh, I pushed the branch to the main repo by mistake. I'll open a new pull request with the latest version.

**Pull request closed without merging**